src/go/rdctl/pkg/snapshot/manager.go

```diff
@@ -139,6 +156,12 @@ func (manager Manager) Delete(id string) error {
 		return fmt.Errorf("snapshot with id %q does not exist", id)
 	}
 	snapshotDir := filepath.Join(manager.Paths.Snapshots, id)
+	// Remove complete.txt file. This must be done first because restoring
+	// from a partially-deleted snapshot could result in errors.
+	completeFilePath := filepath.Join(snapshotDir, completeFileName)
```

Comment:
The whole block at the beginning of this method can be deleted because it is only called with an id returned by `GetSnapshotId`.

```golang
	dirEntries, err := os.ReadDir(manager.Paths.Snapshots)
	if err != nil {
		return fmt.Errorf("failed to read snapshots dir: %w", err)
	}
	found := false
	for _, dirEntry := range dirEntries {
		if dirEntry.Name() == id {
			found = true
			break
		}
	}
	if !found {
		return fmt.Errorf("snapshot with id %q does not exist", id)
	}
```

If you really want to validate again (which I don't think is necessary), then you can make this change:

```suggestion
	completeFilePath := filepath.Join(snapshotDir, completeFileName)
	if _, err := os.Stat(completeFilePath); err != nil {
		return fmt.Errorf("snapshot with id %q does not exist", id)
	}
```